binding.cc

```diff
@@ -1179,6 +1180,26 @@ namespace zmq {
     NanReturnValue(NanNew<String>(version_info));
   }
 
+#if ZMQ_VERSION_MAJOR >= 4
+   static NAN_METHOD(ZmqCurveKeypair) {
+    NanScope();
+
+    char public_key [41];
+    char secret_key [41];
+
+    int rc = zmq_curve_keypair( public_key, secret_key);
+    if (rc < 0) {
+      throw std::runtime_error(ErrorMessage());
```

Comment:
eh ya now that @ronkorving mentions it, i don't like this `std::runtime_error` at all. that's just another obscure and confusing C++ convention IMO. 

it's always tricky to throw errors, but ideally what's raised clues us at solution. in JavaScript, we expect stuff about context/stack calls.

Running the `curve_keypair` operation is self-determined and constrained, *no function parameters given from JS land: <strong>only output</strong>*; therefore, the likelihood is libzmq version or lack of libsodium.

While it could be an unsafe assumption about the program error, we're better off being descriptive about what is likely and possible, so what do you think of throwing something like: 

> `zmq_curve_keypair()` *libzmq v4+ -with-libsodium* operation failed.